spec/codecs/protobuf3_spec.rb

```diff
@@ -0,0 +1,269 @@
+        :appliedPolicyType => PBDNSMessage::PolicyType::NSIP}
+
+      dns_rr_data = [
+        {:name => "abc", :type => 9000, :class => 8000, :ttl => 20, :rdata => "300"},
+        {:name => "def", :type => 19000, :class => 18000, :ttl => 120, :rdata => "1300"}
+      ]
+
+      dns_response_data[:rrs] = dns_rr_data.map { | d | d = dns_rr_class.new(d) }
+      dns_response_object = dns_response_class.new(dns_response_data)
+
+      pbdns_message_data = {
+        # :UUID => '12345678901233456789', :TaskPingIPv4Result => ping_result_object
+        :type => PBDNSMessage::Type::DNSIncomingResponseType,
+        :messageId => "15",
+        :serverIdentity => "16",
+        :socketFamily => PBDNSMessage::SocketFamily::INET6,
+        :socketProtocol => PBDNSMessage::SocketProtocol::TCP,
+        :from => "17",
+        :to => "18",
+        :inBytes => 70000,
+        :timeSec => 80000,
+        :timeUsec => 90000,
+        :id => 20000,
+        :question => dns_question_object,
+        :response => dns_response_object,
+        :originalRequestorSubnet => "19",
+        :requestorId => "Bar",
+        :initialRequestId => "20",
+        :deviceId => "21",
+        }
+      pbdns_message_object = pbdns_message_class.new(pbdns_message_data)
+      bin = pbdns_message_class.encode(pbdns_message_object)
+      plugin_4.decode(bin) do |event|
+        
+        ['messageId', 'serverIdentity','from','to','inBytes','timeUsec','timeSec','id', 'originalRequestorSubnet', 'requestorId' ,'initialRequestId','deviceIdf'].each { |n|  
+          expect(event.get(n)).to eq(pbdns_message_data[n.to_sym] ) }
+
+        # enum test: 
+        expect(event.get("type") ).to eq("DNSIncomingResponseType" )
+        expect(event.get("socketFamily") ).to eq("INET6" )
+        expect(event.get("socketProtocol") ).to eq("TCP" )
+
+        expect(event.get("question")["qName"] ).to eq(dns_question_data[:qName] )
+        expect(event.get("question")["qType"] ).to eq(dns_question_data[:qType] )
+        expect(event.get("question")["qClass"] ).to eq(dns_question_data[:qClass] )
+    
+        ['rcode', 'appliedPolicy','tags','queryTimeSec','queryTimeUsec'].each { |n|   expect(event.get('response')[n]).to eq(dns_response_data[n.to_sym] )   }
+        expect(event.get("response")['appliedPolicyType'] ).to eq("NSIP" )
+
+        dns_rr_data.each_with_index { | data, index | 
+          found = event.get("response")['rrs'][index]
+          ['name', 'type','class','ttl','rdata'].each { |n|   expect(found[n]).to eq(data[n.to_sym])   }
+        }
+
+      end
+    end # it
+
+#### Test case 5: decode test case for github issue 17 ####################################################################################################################
+    let(:plugin_5) { LogStash::Codecs::Protobuf.new("class_name" => "com.foo.bar.IntegerTestMessage", "include_path" => [pb_include_path + '/pb3/integertest_pb.rb'], "protobuf_version" => 3)  }
```

Comment:
File `/pb3/integertest_pb.rb` seems to be missing from this PR, causing the specs to fail

Reply:
Oopsie. Added it just now, thanks!